Examples/destination_plugins/MixpanelDestination.swift

```diff
@@ -0,0 +1,302 @@
+            analytics?.log(message: "Mixpanel track \(finalEventName) properties \(String(describing: event.properties?.dictionaryValue))")
+        } else if settings?["trackNamedPages"] as? Bool ?? false, let eventName = event.name {
+            let finalEventName = "Viewed \(eventName) Screen"
+            mixpanelTrack(finalEventName, properties: event.properties?.dictionaryValue)
+            analytics?.log(message: "Mixpanel track \(finalEventName) properties \(String(describing: event.properties?.dictionaryValue))")
+        } else if settings?["trackCategorizedPages"] as? Bool ?? false, let category = event.category {
+            let finalEventName = "Viewed \(category) Screen"
+            mixpanelTrack(finalEventName, properties: event.properties?.dictionaryValue)
+            analytics?.log(message: "Mixpanel track \(finalEventName) properties \(String(describing: event.properties?.dictionaryValue))")
+        }
+        
+        return event
+    }
+    
+    func group(event: GroupEvent) -> GroupEvent? {
+        
+        guard let groupID = event.groupId, !groupID.isEmpty,
+              let groupIdentifierProperties = settings?["groupIdentifierTraits"] as? [String] else {
+            return event
+        }
+        
+        // No need to continue if we don't have any properties
+        if groupIdentifierProperties.isEmpty {
+            return event
+        }
+        
+        for key in groupIdentifierProperties {
+            let nsGroupID = NSString(string: groupID)
+            
+            if let traits = event.traits?.dictionaryValue as? Properties {
+                if let group = traits[key] as? String {
+                    mixpanel?.getGroup(groupKey: group, groupID: nsGroupID).setOnce(properties: traits)
+                }
+            }
+            
+            mixpanel?.setGroup(groupKey: key, groupID: nsGroupID)
+            analytics?.log(message: "Mixpanel setGroup \(key) groupID \(groupID)")
+        }
+        
+        return event
+    }
+    
+    func alias(event: AliasEvent) -> AliasEvent? {
+        // Use Mixpanel generated id
+        if let distinctId = mixpanel?.distinctId, let newId = event.userId {
+            mixpanel?.createAlias(newId, distinctId: distinctId)
+        }
+        return event
+    }
+    
+    func registeredForRemoteNotificationsWithDeviceToken(_ deviceToken: Data) {
+        mixpanel?.people.addPushDeviceToken(deviceToken)
+        analytics?.log(message: "Mixpanel people addPushDeviceToken \(deviceToken.toString())")
+    }
+    
+    func reset() {
+        flush()
+        
+        mixpanel?.reset()
+        analytics?.log(message: "Mixpanel reset")
+    }
+    
+    func flush() {
+        mixpanel?.flush()
+        analytics?.log(message: "Mixpanel Flush")
+    }
+    
+    // MARK: - Private Methods
+    private func mixpanelTrack(_ eventName: String, properties: [String: Any]? = nil) {
```

Comment:
might be good to move non-protocol conformance methods out into an extension.

Reply:
Done